src/index.js

```diff
@@ -40,40 +41,72 @@ function glamorous(comp, {rootEl, displayName} = {}) {
      *   merged predicably (with no regard to specificity)
      * @return {ReactElement} React.createElement
      */
-    function GlamorousComponent(props) {
-      const {className, ...rest} = props
-      const {toForward, cssOverrides} = splitProps(rest, GlamorousComponent)
-
-      // create className to apply
-      const mappedArgs = GlamorousComponent.styles.map(mapToPropsCall)
-      const {
-        glamorStyles: parentGlamorStyles,
-        glamorlessClassName,
-      } = extractGlamorStyles(className)
-      const glamorClassName = css(
-        ...mappedArgs,
-        ...parentGlamorStyles,
-        cssOverrides,
-      ).toString()
-      const fullClassName = joinClasses(glamorlessClassName, glamorClassName)
-
-      return React.createElement(GlamorousComponent.comp, {
-        className: fullClassName,
-        ...toForward,
-      })
-
-      function mapToPropsCall(glamorRules) {
-        if (typeof glamorRules === 'function') {
-          return glamorRules(props)
+    class GlamorousComponent extends Component {
+      constructor(props, context) {
+        super(props, context)
+        this.state = {
+          theme: null,
         }
-        return glamorRules
+        this.setTheme = this.setTheme.bind(this)
+      }
+
+      setTheme(theme) {
+        this.setState({
+          theme,
+        })
+      }
+
+      componentWillMount() {
```

Comment:
The advantage of using `componentDidMount` here is that the `setState` call in `setTheme` will not cause a re-render after the initial render.